site/oauth2.md

```diff
@@ -0,0 +1,464 @@
+ * `<name_pattern>` is a wildcard pattern for resource name
+ * `&lt;routing_key_pattern>` is an optional wildcard pattern for routing key in topic authorization
+
+Wildcard patterns are strings with optional wildcard symbols `*` that match
+any sequence of characters.
+
+Wildcard patterns match as following:
+
+ * `*` matches any string
+ * `foo*` matches any string starting with a `foo`
+ * `*foo` matches any string ending with a `foo`
+ * `foo*bar` matches any string starting with a `foo` and ending with a `bar`
+
+There can be multiple wildcards in a pattern:
+
+ * `start*middle*end`
+ * `*before*after*`
+
+**To use special characters like `*`, `%`, or `/` in a wildcard pattern,
+the pattern must be [URL-encoded](https://en.wikipedia.org/wiki/Percent-encoding).**
+
+These are the typical permissions examples:
+
+- `read:*/*`(`read:*/*/*`) - read permissions to any resource on any vhost
+- `write:*/*`(`write:*/*/*`) - write permissions to any resource on any vhost
+- `read:vhost1/*`(`read:vhost1/*/*`) - read permissions to any resource on the `vhost1` vhost
+- `read:vhost1/some*` - read permissions to all the resources, starting with `some` on the `vhost1` vhost
+- `write:vhsot1/some*/routing*` - topic write permissions to publish to an exchange starting with `some` with a routing key starting with `routing`
+
+See the [wildcard matching test suite](./test/wildcard_match_SUITE.erl) and [scopes test suite](./test/scope_SUITE.erl) for more examples.
+
+Scopes should be prefixed with `resource_server_id`. For example,
+if `resource_server_id` is "my_rabbit", a scope to enable read from any vhost will
+be `my_rabbit.read:*/*`.
+
+### <a id="use-different-token-field" class="anchor" href="#use-different-token-field">Using a different token field for the Scope</a>
+
+By default the plugin will look for the `scope` key in the token, you can configure the plugin to also look in other fields using the `extra_scopes_source` setting. Values format accepted are scope as **string** or **list**
+
+
+<pre class="lang-erlang">
+[
+  {rabbitmq_auth_backend_oauth2, [
+    {resource_server_id, &lt;&lt;"my_rabbit_server">>},
+    {extra_scopes_source, &lt;&lt;"my_custom_scope_key">>},
+    ...
+    ]}
+  ]},
+].
+</pre>
+Token sample:
+<pre class="lang-ini">
+{
+ "exp": 1618592626,
+ "iat": 1618578226,
+ "aud" : ["my_id"],
+ ...
+ "scope_as_string": "my_id.configure:*/* my_id.read:*/* my_id.write:*/*",
+ "scope_as_list": ["my_id.configure:*/*", "my_id.read:*/*", my_id.write:*/*"],
+ ...
+ }
+</pre>
+
+### <a id="use-tokens-with-clients" class="anchor" href="#use-tokens-with-clients">Using Tokens with Clients</a>
+
+A client must present a valid `access_token` acquired from an OAuth 2.0 provider (UAA) as the **password**
+in order to authenticate with RabbitMQ.
+
+To learn more about UAA/OAuth 2.0 clients see [UAA docs](https://github.com/cloudfoundry/uaa/blob/master/docs/UAA-APIs.rst#id73).
```

Comment:
To learn more about OAuth 2.0 clients, see the [OAuth 2.0 client specification](https://www.rfc-editor.org/rfc/rfc6749#section-4.4).